Add tests of improved validation error messages for oneOf

Bjwebb opened the pull request:
openownership/cove-bods#16

Depends on this lib-cove PR https://github.com/OpenDataServices/lib-cove/pull/12

odscjames commented:
badfile_all_validation_errors.json seems to be a new file but I can't see it actually being used anywhere - have I missed something?

I would put the upgrading of lib-cove to 0.5 in the changelog - we have to make sure that requirement is meet upstream when upgrading openownership/cove-bods and it would be good if it was clear in the Changelog. Also, that produces behaviour change in this repo in the types of error messages approved, it seems worth noting.

Bjwebb commented:
> badfile_all_validation_errors.json seems to be a new file but I can't see it actually being used anywhere - have I missed something?

Sorry, this is a holdover from the changes I removed from this branch. I have removed it.

> I would put the upgrading of lib-cove to 0.5 in the changelog

Done.
